### DISCUSSION TITLE
React native mutation error catching

### DISCUSSION THREAD

**Discussion by olup** [TanStack/query#670] [General] [answered]
Not sure that it's not my way of handling things but :
I have an axios method that will throw on error.
I have a useMutation that has `throwOnError : true`.
I can catch the error alright when I call my mutation (and logging it in the console).

However, there is still an unhandled promise that breaks my React Native app when I call the mutation on some error prone route.

Any idea why? If I use my fetcher function directly I do not have this problem.

Oh, and I am running latest v.2

**Comment by tannerlinsley** [comment, marked as answer]
Yep. You need to set the logger to be console.log for errors, not console.error. Redbox error display on RN pretty aggressive.

https://github.com/tannerlinsley/react-query

**Comment by olup** [comment]
Oh, i missed that ! Cheers for a quick reply !

Le mar. 30 juin 2020 à 16:52, Tanner Linsley <notifications@github.com> a
écrit :
